lib/mayaUsd/resources/ae/usd-shared-components/src/python/usdSharedComponents/collection/includeExcludeWidget.py

```diff
@@ -110,114 +106,40 @@ def __init__(
-
-        self._include.cbIncludeAll.setChecked(
-            self._collection.GetIncludeRootAttr().Get()
-        )
-
-        self._updatingUI = False
+    def _onDataChanged(self):
+        incAll = self._collData.includesAll()
+        if incAll != self._include.cbIncludeAll.isChecked():
+            self._include.cbIncludeAll.setChecked(incAll)
 
     def onAddToIncludePrimClicked(self):
-        pickedItems: Sequence[Usd.Prim] = Host.instance().pick(self._prim.GetStage(), dialogTitle="Add Include Objects")
-        if pickedItems is None:
-            return
-        self._updatingUI = True
-        for item in pickedItems:
-            self._collection.GetIncludesRel().AddTarget(item.GetPath())
-        self._updatingUI = False
-        self.updateUI()
+        items = self._collData.pick("Add Include Objects")
+        self._collData.getIncludeData().addStrings(items)
 
     def onAddToExcludePrimClicked(self):
-        pickedItems: Sequence[Usd.Prim] = Host.instance().pick(self._prim.GetStage(), dialogTitle="Add Exclude Objects")
-        if pickedItems is None:
-            return
-        self._updatingUI = True
-        for item in pickedItems:
-            self._collection.GetExcludesRel().AddTarget(item.GetPath())
-        self._updatingUI = False
-        self.updateUI()
+        items = self._collData.pick("Add Exclude Objects")
+        self._collData.getExcludeData().addStrings(items)
 
     def onRemoveSelectionFromInclude(self):
-        self._updatingUI = True
-        for item in self._include.list.selectedItems:
-            self._collection.GetIncludesRel().RemoveTarget(Sdf.Path(item))
-        self._updatingUI = False
-        self.updateUI()
+        self._collData.getIncludeData().removeStrings(self._include.list.selectedItems())
         self.onListSelectionChanged()
 
     def onRemoveSelectionFromExclude(self):
-        self._updatingUI = True
-        for item in self._exclude.list.selectedItems:
-            self._collection.GetExcludesRel().RemoveTarget(Sdf.Path(item))
-        self._updatingUI = False
-        self.updateUI()
+        self._collData.getExcludeData().removeStrings(self._include.list.selectedItems())
         self.onListSelectionChanged()
 
     def onListSelectionChanged(self):
-        includesSelected = self._include.list.hasSelectedItems
-        excludeSelected = self._exclude.list.hasSelectedItems
+        includesSelected = self._include.list.hasSelectedItems()
+        excludeSelected = self._exclude.list.hasSelectedItems()
         self._deleteBtn.setEnabled(includesSelected or excludeSelected)
 
-        self._deleteBtnActionFromIncludes.setEnabled(includesSelected)
-        self._deleteBtnActionFromExcludes.setEnabled(excludeSelected)
+        # TODO: these QAction are reported as already being deleted in C++
+        # self._deleteBtnActionFromIncludes.setEnabled(includesSelected)
+        # self._deleteBtnActionFromExcludes.setEnabled(excludeSelected)
```

Comment:
Note: these line fails in Maya, even in the current dev branch without my changes. Qt complains that the C++ QAction object has already been deleted. I think there is an issue with the retention of the QAction.

Reply:
-- whild guess begin -- 

maybe creating the QMenu in line 64 with self as a parent object may help with the retention of the menu?

-- whild guess end --

Reply:
I found a way around the problem: don't try to keep the action objects as variable, look them up when needed. I suppose Qt destroys and rebuilds them? I know Maya create menus dynamically, maybe it is due to that.